### ISSUE TITLE
AWS API rate exceeded - operations failing blindly

### ISSUE THREAD

**Issue by Alfaj0r** [awslabs/amazon-dynamodb-tools#32]
I just ran table_tagger.py against a few hundred DynamoDB tables. The console output made me feel like a real DevOps champion, having logs that every single one of my tables was tagged.

However, I reviewed CloudTrail logs and found that my work had not happened as fatefully as I had thought. Almost half of the calls logged some sort of error...
![image](https://github.com/awslabs/amazon-dynamodb-tools/assets/1396050/1e9b807b-94c0-4e5d-97ee-1c3c0cefa166)

Drilling down for all the errors, they are all the same:
![image](https://github.com/awslabs/amazon-dynamodb-tools/assets/1396050/42d6c446-8a6c-4e6f-821d-68f4cec81d09)
![image](https://github.com/awslabs/amazon-dynamodb-tools/assets/1396050/0b024675-c32a-4de0-ab5c-09934ba9d032)

The script needs to be enhanced so that:

1. it does not log successes for API calls that were not successfully executed
2. it does not exceed the rate limits for the AWS API

**Comment by LeeroyHannigan** [comment]
Is your requirement for tagging the tables for granular cost in Cost Explorer, if so, there is no longer a need to do this. I'll update the READ ME later this week.

**Comment by Alfaj0r** [comment]
Hey @LeeroyHannigan thanks for taking a look.
You might be thinking about the available "Resource" extra breakdown in cost-explorer, but that's only good for 2 weeks...
![image](https://github.com/awslabs/amazon-dynamodb-tools/assets/1396050/0e8838d3-4786-426d-9757-9642d6dc7f01)

I want this to run for longer-term analysis too.

Is this what you were thinking about?

**Comment by LeeroyHannigan** [comment]
Yes exactly what I was thinking about. We need to take in the throttling for tagging resources, once I have some bandwidth I'll take a look.
